crates/builder/src/tasks/tx_poller.rs

```diff
@@ -0,0 +1,110 @@
+use std::sync::{Arc, Mutex};
+use std::time::Duration;
+use std::{collections::HashMap, time};
+
+use alloy_consensus::TxEnvelope;
+use eyre::Error;
+use reqwest::Client;
+use serde_json::from_str;
+use serde_json::Value;
+use tokio::sync::mpsc;
+use tokio::task::JoinHandle;
+
+pub use crate::config::BuilderConfig;
+
+/// Implements a poller for the block builder to pull transactions from the transaction pool.
+pub struct TxPoller {
+    // config for the builder
+    pub config: BuilderConfig,
+    // Reqwest client for fetching from the tx-pool
+    pub client: Client,
+    //  Maintain a set of txn hash to expiration time
+    pub seen_txns: Arc<Mutex<HashMap<String, time::Instant>>>,
```

Comment:
instead of string key, use `B256` key so the tx hash can be used directly

Reply:
I'll do you one better and make it a `TxHash` which is a type alias for B256 and let's us just use `.tx_hash()` in the `entry` call. 🎉